### ISSUE TITLE
Proposal: do everything in one go

### ISSUE THREAD

**Issue by lig** [matthiask/html-sanitizer#2]
It looks like html-sanitizer does a lot of wibbly-wobbly things to produce the result.

It seems possible to implement every html-sanitizer feature using a smart plugin system for a custom SAX parser rules realization.

**Comment by lig** [comment]
I'm going to come back here and implement this when I have enough time.

**Comment by matthiask** [comment]
Yes, this should be possible. The core of the cleaner is now more than 8 years old, so I probably would do it differently too these days :-)
https://github.com/feincms/feincms/commit/0ec8e834dd2e0927bb23d46ee9102716c7735add

Luckily we have a testsuite now to catch regressions. However, it is not very exhaustive yet, so maybe it would also be good to expand the testsuite before moving to big refactorings. (No must-have at all, but would certainly make reviewing eventual pull requests easier!)
